pkg/controller/migplan/migplan_controller.go

```diff
@@ -389,43 +391,46 @@ func (r *ReconcileMigPlan) planSuspended(plan *migapi.MigPlan) error {
 
 func (r *ReconcileMigPlan) AddRunningLabelForPlan(plan *migapi.MigPlan) error {
 	migrations, err := plan.ListMigrations(r)
+	runningMigrations := 0
+	needsUpdate := false
 	if err != nil {
 		return liberr.Wrap(err)
 	}
 
-	if len(migrations) == 0 {
+	//Check if there are any associated migrations that are running
+	for _, m := range migrations {
+		if m.Status.HasCondition(migctl.Running) {
+			runningMigrations++
+		}
+	}
+
+	//No migrations associated with the migplan or there are no running migrations for the Migplan
+	if len(migrations) == 0 || runningMigrations == 0 {
+		if plan.Labels != nil {
```

Comment:
I am curious if this check needed? Labels field should always be present on migplan right?

Reply:
Actually migplan does not have any labels associated, it just has the annotations field.

Reply:
I thought even if there are no labels, Labels field will be allocated by the decoder and hence always be non-nil

Reply:
let's keep it anyway to safe guard againt panics, +1